collector/drbd_linux.go

```diff
@@ -0,0 +1,213 @@
+// Copyright 2016 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/log"
+)
+
+// Numerical metric provided by /proc/drbd.
+type drbdNumericalMetric struct {
+	desc       *prometheus.Desc
+	valueType  prometheus.ValueType
+	multiplier float64
+}
+
+func newDrbdNumericalMetric(name string, desc string, valueType prometheus.ValueType, multiplier float64) drbdNumericalMetric {
+	return drbdNumericalMetric{
+		desc: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, "drbd", name),
+			desc,
+			[]string{"device"}, nil),
+		valueType:  valueType,
+		multiplier: multiplier,
+	}
+}
+
+// String pair metric provided by /proc/drbd.
+type drbdStringPairMetric struct {
+	desc      *prometheus.Desc
+	valueOkay string
+}
+
+func (metric *drbdStringPairMetric) isOkay(value string) float64 {
+	if value == metric.valueOkay {
+		return 1
+	}
+	return 0
+}
+
+func newDrbdStringPairMetric(name string, desc string, valueOkay string) drbdStringPairMetric {
+	return drbdStringPairMetric{
+		desc: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, "drbd", name),
+			desc,
+			[]string{"device", "node"}, nil),
+		valueOkay: valueOkay,
+	}
+}
+
+var (
+	drbdNumericalMetrics = map[string]drbdNumericalMetric{
+		"ns": newDrbdNumericalMetric(
+			"network_sent_bytes",
```

Comment:
Adding the `_total` suffix: yes! That sounds like a good idea.

Regarding changing the documentation string: I made sure that all of the documentation strings are literal copies of how they are [officially documented](https://www.drbd.org/en/doc/users-guide-84/ch-admin#s-performance-indicators), except that I dropped the "in Kibyte" part. I've now restored those parts to say "in bytes".

To summarize: the strings are now the same as upstream `| sed -e 's/Kibyte/bytes/'`